### ISSUE TITLE
Codification maps not removable

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap--Build-Your-Own-Blocks#1886]
When I defined a map and wanted to later remove it I tried mapping to the empty box and the old mapping still applied. A workaround is to map to the space character.

Also I found it surprising that maps were part of the state of a project. I expected that I needed to recreate the maps when the green flag is clicked. Instead all maps are preserved in the state of the project.

**Comment by jmoenig** [comment]
Ah yes, it cost me a lot of effort to preserve manually created codification maps in the project state!

Codification maps can also be entered via the GUI when "codification support" is enabled. Then you can right-click on blocks and codifiable syntax elements to manually map them to a textual encoding using a dialog box. I use this a lot when demoing the feature, and find it easier to explain when I (rant against the idea of blocks not being "real") first introduce people to this idea.

You can also remove a mapping through the GUI. I agree it's kinda cumbersome, and - honestly - I heven't previously thought about removing all maps. Should we add a menu-entry for this?

**Comment by ToonTalk** [comment]
Makes sense to persist the mappings when created by the GUI and then of course the mapping blocks shouldn't be any different. Still think that the map block should remove the mapping if the code box is empty.

I did need to delete all mappings so I saved all the blocks and imported them into a new project.

**Comment by ToonTalk** [comment]
By the way, in Chrome Windows 10 when editing the code mapping via the GUI I can't select text (either via control-a or via the mouse). I can only move the cursor.
